Flaky test: IngressITTest.ingressAnnotations times out on OCP 4.16

## Description

`IngressITTest.ingressAnnotations` fails intermittently on OCP 4.16 with a 30-second
awaitility timeout. The operator fails to create the Ingress resource within the timeout
window. The same test passes reliably on OCP 4.22 (Latest).

## Environment

- **Registry version:** 3.3.1
- **Operator image:** `quay.io/apicurio/apicurio-registry-3-operator:3.3.1`
- **OCP version:** 4.16.55 (Fyre-provisioned EBC cluster)
- **Test run:** CSD pipeline `Controller smoke+feature (OCP 4.16)`

## Test Results

- **OCP Latest (4.22.2):** 39/39 passed — `ingressAnnotations` passed
- **OCP 4.16 (4.16.55):** 38/39 passed — `ingressAnnotations` failed

## Error

```
[ERROR] io.apicurio.registry.operator.it.IngressITTest.ingressAnnotations
        -- Time elapsed: 36.49 s <<< ERROR!

org.awaitility.core.ConditionTimeoutException:
  Assertion condition defined as a Lambda expression in
  io.apicurio.registry.operator.it.IngressITTest null within 30 seconds.
    at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:167)
    at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:790)
    at io.apicurio.registry.operator.it.IngressITTest.ingressAnnotations(IngressITTest.java:48)

Caused by: java.lang.IllegalStateException: Kubernetes resource does not exist.
    at io.apicurio.registry.operator.utils.K8sCell.get(K8sCell.java:63)
    at io.apicurio.registry.operator.it.IngressITTest.lambda$ingressAnnotations$3(IngressITTest.java:49)
```

The test waited 30 seconds for the Ingress resource to be created by the operator, but
the resource never appeared.

## Contributing Factors

### 1. K8s 409 Conflict retries during the preceding test

The `ingressClassName` test (which passed) triggered repeated 409 Conflict errors on the
custom resource, indicating the operator and test were racing to update the same CR:

```
Retrying update of ResourceID{name='ingress-class-name', namespace='test-8c55a46'}
because of a Kubernetes client exception:
  Operation cannot be fulfilled on apicurioregistries3.registry.apicur.io "ingress-class-name":
  the object has been modified; please apply your changes to the latest version and try again.
```

This happened at least twice (15:27:51 and 15:28:03), suggesting reconciliation contention
that may have left the operator in a state where it was slow to process the next CR
(`ingress-annotations`).

### 2. Operator startup probe failure

The operator pod itself experienced a startup probe failure earlier in the test namespace
lifecycle:

```
Startup probe failed: Get "http://10.254.20.68:8080/q/health/started":
  dial tcp 10.254.20.68:8080: connect: connection refused
```

This suggests the operator was slower to become ready on this OCP 4.16 cluster.

### 3. Cluster-wide image pull pressure

The OCP 4.16 cluster had `pull QPS exceeded` errors across several OpenShift system
components (cloud-credential-operator, controller-manager-operator, insights-operator,
kube-scheduler-operator). While these are unrelated to the Apicurio operator itself, they
indicate the cluster was under resource pressure which may have contributed to slower
reconciliation.

## Analysis

The `ingressAnnotations` test creates an `ApicurioRegistry3` CR and waits up to 30 seconds
for the operator to create an Ingress resource (`IngressITTest.java:48-49`). On this
OCP 4.16 cluster, the operator reconciliation was slower — likely due to a combination of
409 Conflict retries from the preceding test and general cluster pressure — causing the
Ingress to not be created within the timeout window.

The `ingressClassName` test in the same class passed, so the operator itself is functional —
this is a timing-sensitive test that doesn't account for slower clusters.

## Possible Fixes

1. **Increase the awaitility timeout** for `ingressAnnotations` (e.g., 60-90 seconds
   instead of 30) to tolerate slower clusters
2. **Add explicit CR cleanup verification** between tests to ensure the operator has fully
   reconciled before starting the next test
3. **Add retry logic** in the test for the case where the Ingress doesn't exist yet,
   distinguishing between "not yet created" and "will never be created"

--- a/operator/controller/src/test/java/io/apicurio/registry/operator/it/IngressITTest.java
+++ b/operator/controller/src/test/java/io/apicurio/registry/operator/it/IngressITTest.java
@@ -45,12 +45,17 @@ public class IngressITTest extends ITBase {
         final var appIngress = k8sCell(client, () -> client.network().v1().ingresses().withName(primary.getCached().getMetadata().getName() + "-app-ingress").get());
         final var uiIngress = k8sCell(client, () -> client.network().v1().ingresses().withName(primary.getCached().getMetadata().getName() + "-ui-ingress").get());
 
-        await().atMost(SHORT_DURATION).ignoreExceptions().untilAsserted(() -> {
-            assertThat(appIngress.get()).isNotNull();
+        // Creating the Ingress requires a full reconciliation of the freshly created CR, which can
+        // take considerably longer than a subsequent update to an already existing resource. Use
+        // MEDIUM_DURATION here, consistent with the other resource-creation waits such as
+        // ITBase.checkDeploymentExists(). getOptional() is used instead of get() so that a timeout
+        // reports the resource as missing rather than surfacing an IllegalStateException.
+        await().atMost(MEDIUM_DURATION).ignoreExceptions().untilAsserted(() -> {
+            assertThat(appIngress.getOptional()).isPresent();
         });
 
-        await().atMost(SHORT_DURATION).ignoreExceptions().untilAsserted(() -> {
-            assertThat(uiIngress.get()).isNotNull();
+        await().atMost(MEDIUM_DURATION).ignoreExceptions().untilAsserted(() -> {
+            assertThat(uiIngress.getOptional()).isPresent();
         });
 
         await().atMost(SHORT_DURATION).ignoreExceptions().untilAsserted(() -> {

